esmvaltool/utils/testing/regression/compare.py

```diff
@@ -0,0 +1,398 @@
+    elif not np.array_equal(ref, cur):
+        if np.issubdtype(ref.dtype, np.inexact) and np.issubdtype(
+                cur.dtype, np.inexact) and np.allclose(ref, cur):
+            msg.append("data is almost but not quite the same")
+        else:
+            msg.append("data is different")
+    return as_txt(msg)
+
+
+def diff_dataarray(ref: xr.DataArray, cur: xr.DataArray, type_: str) -> str:
+    """Compare two xarray DataArrays and describe the difference."""
+    msg = []
+    if not cur.identical(ref):
+        msg.append(f"{type_} '{cur.name}' is not identical to reference")
+    if diff := diff_array(ref.values, cur.values):
+        msg.append(diff)
+    if diff := diff_attrs(ref.attrs, cur.attrs):
+        msg.append(indent(diff, "  "))
+    return as_txt(msg)
+
+
+def diff_dataset(ref: xr.Dataset, cur: xr.Dataset) -> str:
+    """Compare two xarray Datasets and describe the difference."""
+    msg = []
+    if diff := diff_attrs(ref.attrs, cur.attrs):
+        msg.append(diff)
+
+    for var in ref:
+        if var not in cur:
+            msg.append(f"missing variable '{var}'")
+        else:
+            if diff := diff_dataarray(ref[var], cur[var], "variable"):
+                msg.append(diff)
+
+            for coord in ref[var].coords:
+                if coord not in cur.coords:
+                    msg.append(f"missing coordinate '{coord}'")
+                elif diff := diff_dataarray(ref[var].coords[coord],
+                                            cur[var].coords[coord],
+                                            'coordinate'):
+                    msg.append(diff)
+            for coord in cur[var].coords:
+                if coord not in ref.coords:
+                    msg.append(f"extra coordinate '{coord}'")
+
+    for var in cur:
+        if var not in ref:
+            msg.append(f"extra variable '{var}'")
+
+    return '\n'.join(msg)
+
+
+def load_nc(filename: Path) -> xr.Dataset:
+    """Load a NetCDF file."""
+    dataset = xr.open_dataset(filename, chunks={}, decode_times=False)
+    for attr in IGNORE_GLOBAL_ATTRIBUTES:
+        dataset.attrs.pop(attr, None)
+    return dataset
+
+
+def debug_nc(reference_file: Path, current_file: Path) -> str:
+    """Find out the differences between two NetCDF files."""
+    ref = load_nc(reference_file)
+    cur = load_nc(current_file)
+
+    if diff := diff_dataset(ref, cur):
+        msg = diff
+    else:
+        msg = "Unknown difference"
+
+    return msg
+
+
+def debug_txt(reference_file: Path, current_file: Path) -> str:
+    """Find out the differences between two text files."""
+    with reference_file.open('rt') as file:
+        ref = file.readlines()
+    with current_file.open('rt') as file:
+        cur = file.readlines()
+
+    msg = difflib.unified_diff(
+        cur,
+        ref,
+        fromfile=str(current_file),
+        tofile=str(reference_file),
+    )
+    return "".join(msg)
+
+
+debug_csv = debug_txt
+debug_html = debug_txt
```

Comment:
No, because the .html files contain scientific output. At least for recipe_impact.yml, the HTML file contains the numbers from the computation. Maybe we could change the diagnostics so they no longer write a date to the file?